File: js/Neetcode_150/4_Binary_Search/21_bs1.js
//BS1: given a sorted array & 'target'; return the index of target in arr; if target not present in arr, return -1.

var search = function(nums, target) {
    let n = nums.length
    let l=0, r = n-1
    while(l<=r){
        let m = Math.floor(l + (r-l)/2)
        if(nums[m] > target) r = m-1
        else if(nums[m] < target) l = m+1
        else return m
    }
    return -1
};

File: c++/Neetcode_150/4_Binary_Search/21_bs1.cpp
//BS1: given a sorted array & 'target'; return the index of target in arr; if target not present in arr, return -1.

int search(vector<int>& nums, int target) {
        int n = nums.size();
        int l = 0, r = n-1;
        while(l<=r){
            int m = l + (r-l)/2;
            if(nums[m] > target) r = m-1;
            else if(nums[m] < target) l = m+1;
            else return m;
        }
        return -1;
    }